pkg/generator/generate.go

```diff
@@ -551,7 +553,14 @@ func (g *schemaGenerator) generateType(
 	if len(t.Type) == 0 {
 		return codegen.EmptyInterfaceType{}, nil
 	}
-	if len(t.Type) != 1 {
+	if len(t.Type) == 2 {
```

Comment:
This assumes `null` is always the second element, and that the non-null type is the first one. I would have to support both.